### DISCUSSION TITLE
SNMP Temperature

### DISCUSSION THREAD

**Discussion by alanjob84** [micropython/micropython#15326] [RP2040 / Pico]
Hello all
I have been given a few PicoW boards and had an idea to turn them into SNMP temperature monitors.
I have been struggling for a few days now trying to get something to work.
I have get them connecting to wifi and reading the onboard temperature.
Where i am failing if getting the temperatore into an SNMP OID to read from an SNMP client.
I can do it without issue in a Linux OS but struggling to find anything for MicroPython.
Any help would be greatly appreciated.
Alan

**Comment by Josverl** [comment]
Possible a start is : https://github.com/PinkInk/upylib/tree/master/usnmp
> Simple Network Management Protocol v1 packet codec, for Micropython.
Codes and decodes SNMP v1 packets received from, or to send on, the network.
Only SNMPv1 is supported. SNMPv1 is not encrypted; data is sent to the network, including the internet if you so choose, 'in the clear'.

but its been mostly inactive for the last years - so you may need to do some updating to todays MicroPython

**Reply by alanjob84** [reply]
Thank you for the reply.

I have looked at usnmp but cannot get my head round its coding.
At the moment I am running the data into an html page and have my monitoring software reading it from there.

Alan
